name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/FidelityPDFExtractor.java

```diff
@@ -0,0 +1,143 @@
+package name.abuchen.portfolio.datatransfer.pdf;
+
+import name.abuchen.portfolio.datatransfer.ExtractorUtils;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.Block;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.DocumentType;
+import name.abuchen.portfolio.datatransfer.pdf.PDFParser.Transaction;
+import name.abuchen.portfolio.model.BuySellEntry;
+import name.abuchen.portfolio.model.Client;
+import name.abuchen.portfolio.model.PortfolioTransaction;
+import name.abuchen.portfolio.model.Transaction.Unit;
+import name.abuchen.portfolio.money.Money;
+import name.abuchen.portfolio.money.Values;
+
+public class FidelityPDFExtractor extends AbstractPDFExtractor
+{
+    public FidelityPDFExtractor(Client client)
+    {
+        super(client);
+
+        addBankIdentifier("FIDELITY BROKERAGE SERVICES LLC"); //$NON-NLS-1$
+
+        addBuySellTransaction();
+        // addSellTransaction();
+        // addDividendTransaction();
+        // addInboundDelivery();
+    }
+
+    @SuppressWarnings("nls")
+    private void addBuySellTransaction()
+    {
+        // line must start with this!
+        DocumentType type = new DocumentType("Transaction Confirmation");
+        this.addDocumentTyp(type);
+
+        Block block = new Block("REFERENCE NO\\. TYPE REG\\.REP\\. TRADE DATE SETTLEMENT DATE CUSIP NO\\. ORDER NO\\.",
+                        "^(?!REFERENCE)(?!DESCRIPTION)[A-Z]{2,5}.*$");
+        type.addBlock(block);
+        block.set(new Transaction<BuySellEntry>()
+
+                        .subject(() -> {
+                            BuySellEntry entry = new BuySellEntry();
+                            entry.setType(PortfolioTransaction.Type.BUY);
+                            return entry;
+                        })
+                        /*
+                         * REFERENCE NO. TYPE REG.REP. TRADE DATE SETTLEMENT
+                         * DATE CUSIP NO. ORDER NO. 23348-SR3JMJ 1* WO# 12-14-23
+                         * 12-18-23 02079K107 23348-I7W2H DESCRIPTION and
+                         * DISCLOSURES You Bought ALPHABET INC CAP STK CL C
+                         * Principal Amount 3,990.00 30 WE HAVE ACTED AS AGENT.
+                         * Settlement Amount 3,990.00 at 133.0000 Symbol: GOOG
+                         */
+
+                        .section("month", "day", "year") //
+                        .match("^\\d+-\\w+.* (?<month>\\d+)-(?<day>\\d+)-(?<year>\\d+) \\d+-\\d+-\\d+.*$")
+                        .assign((t, v) -> //
+                        t.setDate(asDate(v.get("day") + "." + v.get("month") + ".20" + v.get("year"))))
```

Comment:
For example with contributing rules

```suggestion
                        // @formatter:off
                        // 20123-1XXXXX 1* WK# 01-04-21 01-06-21 46428Q109 20123-XXXXX
                        // @formatter:on
                        .section("date") //
                        .match("^.* WK# (?<date>[\\d]{2}\\-[\\d]{2}\\-[\\d]{2}) .*$")
                        .assign((t, v) -> t.setDate(asDate(v.get("date"))))
```